src/foam/dao/history/HistoryRecordServiceServer.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright 2023 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao.history',
+  name: 'HistoryRecordServiceServer',
+  implements: [ 'foam.dao.history.HistoryRecordService' ],
+
+  javaImports: [
+    'foam.core.X',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.mlang.order.Desc',
+    'java.util.stream.Collectors',
+    'java.util.List'
+  ],
+
+  methods: [
+    {
+      name: 'getRecord',
+      type: 'HistoryRecord',
+      args: 'Context x, foam.dao.DAO dao, String propertyName',
+      javaCode: `
+        if ( dao == null ) {
+          throw new IllegalStateException("DAO does not exist");
+        }
+    
+        if ( ! HistoryRecord.class.isAssignableFrom(dao.getOf().getObjClass()) ) {
+          throw new IllegalStateException("DAO must be of HistoryRecord");
+        }
+        
+        ArraySink sink = (ArraySink) dao.orderBy(new Desc(HistoryRecord.TIMESTAMP))
+          .limit(1)
+          .select(new ArraySink());
+        
+        return sink.getArray().size() > 0 ? (HistoryRecord) sink.getArray().get(0) : null;
+      `
+    },
+    {
+      name: 'getRecords',
+      type: 'HistoryRecord[]',
```

Comment:
Is there a reason to not return List instead of array?  You are literally getting a list and creating an array from it

Reply:
I wouldn't think so. I simply tried to match the method return type specified in the ticket but I think it makes more sense to return list as you pointed out. Updated.